free_fleet_adapter/free_fleet_adapter/nav1_robot_adapter.py

```diff
@@ -0,0 +1,553 @@
+        return self.battery_soc
+
+    def get_map_name(self) -> str:
+        return self.map_name
+
+    def get_pose(self) -> Annotated[list[float], 3] | None:
+        transform = self.tf_handler.get_transform()
+        if transform is None:
+            error_message = \
+                f'Failed to update robot [{self.name}]: Unable to get ' \
+                f'transform between base_footprint and map'
+            self.node.get_logger().info(error_message)
+            return None
+
+        orientation = euler_from_quaternion([
+            transform.transform.rotation.x,
+            transform.transform.rotation.y,
+            transform.transform.rotation.z,
+            transform.transform.rotation.w
+        ])
+        robot_pose = [
+            transform.transform.translation.x,
+            transform.transform.translation.y,
+            orientation[2]
+        ]
+        return robot_pose
+
+    def _is_navigation_done(self) -> bool:
+        if self.nav_goal_id is None:
+            return True
+
+        nav_goal_id = self.nav_goal_id
+        goal_status = \
+            self.move_base_handler.get_goal_status(nav_goal_id)
+        if goal_status is None:
+            self.replan_counts += 1
+            self.node.get_logger().error(
+                f'Navigation goal {nav_goal_id} is not found, replan count '
+                f'[{self.replan_counts}]'
+            )
+            self.update_handle.more().replan()
+            return False
+
+        match goal_status.status:
+            case goal_status.PENDING | \
+                 goal_status.ACTIVE | \
+                 goal_status.PREEMPTING | \
+                 goal_status.RECALLING:
+                return False
+            case goal_status.SUCCEEDED:
+                self.node.get_logger().info(
+                    f'Navigation goal {nav_goal_id} reached'
+                )
+                return True
+            case goal_status.PREEMPTED:
+                self.node.get_logger().info(
+                    f'Navigation goal {nav_goal_id} was cancelled'
+                )
+                return True
+            case _:
+                # TODO(ac): test replanning behavior if goal status is
+                # neither executing or succeeded
+                self.replan_counts += 1
+                self.node.get_logger().error(
+                    f'Navigation goal {nav_goal_id} status '
+                    f'{goal_status.status}, replan count '
+                    f'[{self.replan_counts}]'
+                )
+                self.update_handle.more().replan()
+                return False
+
+    def _check_update_handle_initialization(self):
+        if self.update_handle is None:
+            error_message = \
+                f'Failed to update robot {self.name}, robot adapter has ' \
+                'not yet been initialized with a fleet update handle.'
+            self.node.get_logger().error(error_message)
+            raise RuntimeError(error_message)
```

Comment:
In general I would recommend removing this function and checking `update_handle` on a case-by-case basis. For callbacks that are triggered by RMF, there is no need to check `update_handle`. For anything else, throwing an exception is probably too extreme.

Reply:
Removed and returned early in https://github.com/open-rmf/free_fleet/pull/145/commits/cccb4033c024b5035a97c7cda146f70970b24baf